### ISSUE TITLE
Bug: Black Page on load aka Orange Screen of Death

### ISSUE THREAD

**Issue by tigpt** [open-sauced/open-sauced#1006]
### Describe the bug

Sometimes when you are navigating or refreshing the page loads with no content.

### To Reproduce

Didn't isolated the pattern yet, but its easy to reproduce while you navigate / refresh a page.

### Expected behavior

Page should load with content.

### Screenshots

<img width="2160" alt="image" src="https://user-images.githubusercontent.com/43767/118514030-1d1fcf00-b72c-11eb-9d28-9fc9d5ee89bf.png">

### Desktop

- OS: macOS 11.3.1
- Browser: Safari 
- Version: Version 14.1 (16611.1.21.161.6)

### Additional context

Screenshot was taken on page https://opensauced.pizza/repos/open-sauced/open-sauced/1/

**Comment by filiptronicek** [comment]
Thanks for reporting this issue, @tigpt, are there any logs of the console you could share with us (errors, warnings, etc.)?

**Comment by tigpt** [comment]
Yes there are, it looks like `TypeError: null is not an object (evaluating 'n.login')` is the problem.
<img width="1260" alt="image" src="https://user-images.githubusercontent.com/43767/118521454-f749f880-b732-11eb-85be-6e8575be831f.png">

**Comment by bdougie** [comment]
My best guess is that it is related to how the logged user's JWT is store in local storage. 

Reproduction steps:

1. Navigate to a repos page. i.e. https://opensauced.pizza/repos/open-sauced/open-sauced/undefined/
2. Refresh twice consecutively without letting the first refresh complete.

**Comment by mtfoley** [comment]
This may be relevant, but as I look at the codebase (src/index.js and src/components/Nav.js), the only variables that `localStorage.setItem()` or `localStorage.removeItem()` are used with are `isLoggedIn` (boolean I think), `oneGraph:<Config.appId>` (only seeing this removed, but not set), and `adminBar`, which I think is the bar at the top that only org members  see.  @bdougie Should there also be similar function calls for `user` and/or `user.login`?

**Comment by mtfoley** [comment]
I’m not quite to the point of taking the issue but I feel this has as much to do with react error boundaries as anything else.

**Comment by bdougie** [comment]
I saw a tweet from Dan Abramov, this is Suspense acting up.
https://twitter.com/dan_abramov/status/1402935426596364292

going to keep an eye on this discussion here. https://github.com/reactwg/react-18/discussions/37

**Comment by choznerol** [comment]
In my case, I got a HTML response for the JS chunk request and ends up with `Uncaught SyntaxError: expected expression, got '<'`

> <img width="1323" alt="image" src="https://user-images.githubusercontent.com/12410942/140603170-ca11e77b-3227-493b-830a-bbd08ff2b92a.png">

<details><summary> full response HTML </summary>

```html
<!doctype html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><link rel="shortcut icon" href="/favicon.ico"><link rel="manifest" href="/manifest.json"><title>Open Sauced</title><meta name="description" content="The path to your next open-source contribution."><link href="/static/css/main.593f90d4.chunk.css" rel="stylesheet"></head><body><div id="root"></div><script>!function(e){function r(r){for(var n,l,f=r[0],i=r[1],a=r[2],c=0,s=[];c<f.length;c++)l=f[c],Object.prototype.hasOwnProperty.call(o,l)&&o[l]&&s.push(o[l][0]),o[l]=0;for(n in i)Object.prototype.hasOwnProperty.call(i,n)&&(e[n]=i[n]);for(p&&p(r);s.length;)s.shift()();return u.push.apply(u,a||[]),t()}function t(){for(var e,r=0;r<u.length;r++){for(var t=u[r],n=!0,f=1;f<t.length;f++){var i=t[f];0!==o[i]&&(n=!1)}n&&(u.splice(r--,1),e=l(l.s=t[0]))}return e}var n={},o={1:0},u=[];function l(r){if(n[r])return n[r].exports;var t=n[r]={i:r,l:!1,exports:{}};return e[r].call(t.exports,t,t.exports,l),t.l=!0,t.exports}l.m=e,l.c=n,l.d=function(e,r,t){l.o(e,r)||Object.defineProperty(e,r,{enumerable:!0,get:t})},l.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},l.t=function(e,r){if(1&r&&(e=l(e)),8&r)return e;if(4&r&&"object"==typeof e&&e&&e.__esModule)return e;var t=Object.create(null);if(l.r(t),Object.defineProperty(t,"default",{enumerable:!0,value:e}),2&r&&"string"!=typeof e)for(var n in e)l.d(t,n,function(r){return e[r]}.bind(null,n));return t},l.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return l.d(r,"a",r),r},l.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},l.p="/";var f=window.webpackJsonp=window.webpackJsonp||[],i=f.push.bind(f);f.push=r,f=f.slice();for(var a=0;a<f.length;a++)r(f[a]);var p=i;t()}([])</script><script src="/static/js/2.0bb00d63.chunk.js"></script><script src="/static/js/main.da17d60e.chunk.js"></script></body></html>
```

</details>

Can reproduce randomly with `curl` locally:

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/12410942/140603261-b0a72253-1d58-49ad-bc2d-888dc0163f8c.png">

**Comment by 0-vortex** [comment]
Happy to announce this is now fixed by #1322 service worker improvements! 🍕
